src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -105,7 +116,9 @@ public void put(Snapshot.Key key, Delete delete) {
   }
 
   public void to(MutationComposer composer) {
-    if ((composer instanceof PrepareMutationComposer) && isolation == Isolation.SERIALIZABLE) {
+    if ((composer instanceof PrepareMutationComposer)
+        && isolation == Isolation.SERIALIZABLE
+        && strategy == SerializableStrategy.EXTRA_WRITE) {
       toSerializable();
```

Comment:
This method will be refactored in another PR by having `toSerializableWithExtraWrite` and the above if statement will be in the method for consistency with `toSerializableWithExtraRead`.